src/org/labkey/test/tests/core/admin/AllowedFileExtensionAdminTest.java

```diff
@@ -0,0 +1,618 @@
+     * <p>
+     *     After creating a list that has an auto-index and a attachment field as the only field this test will:
+     *     <ul>
+     *         <li>Can insert an element into the list with the allowed file type.</li>
+     *         <li>Validate that a file of the type not allowed is not allowed.</li>
+     *         <li>Remove the bad file and resubmit with a valid file type.</li>
+     *     </ul>
+     * </p>
+     *
+     */
+    @Test
+    public void testAllowedFileExtensionsInLists()
+    {
+
+        List<String> allowedTypes = new ArrayList<>();
+        String excludedType = ".csv";
+
+        for (String extension : fileMap.keySet())
+        {
+            if (!extension.equals(excludedType))
+                allowedTypes.add(extension);
+        }
+
+        log(String.format("Add the following as allowed extensions: %s", allowedTypes));
+        setAllowedExtensions(allowedTypes, allowedTypes);
+
+        goToProjectHome();
+        String listName = "Test Allowed Attachments";
+        String keyField = "Key";
+        String attachmentField = "Attachment Field";
+
+        log(String.format("Create a list '%s' with attachment field and auto-increment key.", listName));
+        _listHelper.createList(getProjectName(), listName, keyField,
+                new FieldDefinition(attachmentField, FieldDefinition.ColumnType.Attachment));
+
+        goToManageLists();
+        _listHelper.goToList(listName);
+
+        String [][] expectedData = new String [allowedTypes.size()][1];
+        int index = 0;
+        for (String allowedType : allowedTypes)
+        {
+            _listHelper.insertNewRow(Map.of(attachmentField, fileMap.get(allowedType).getAbsolutePath()), false);
+            // Add a space before the name to allow for the icon.
+            expectedData[index++][0] = String.format(" %s", fileMap.get(allowedType).getName());
+        }
+
+        goToManageLists();
+        _listHelper.goToList(listName);
+        _listHelper.verifyListData(List.of(new FieldDefinition(attachmentField, FieldDefinition.ColumnType.Attachment)), expectedData, checker());
+
+        _listHelper.goToList(listName);
+        _listHelper.insertNewRow(Map.of(attachmentField, fileMap.get(excludedType).getAbsolutePath()), false);
+
+        // Not sure why we record two exceptions.
+        checkExpectedErrors(2);
+
+        log("Click 'Back' button and select a file type that is allowed.");
+        waitForElement(Locator.button("Back"));
+        clickButton("Back");
+
+        waitForElement(Locator.name("quf_" + attachmentField));
+
+        // Same as Issue 53026, the fields are cleared after hitting back button. Covewrage for that issue is in
+        // testAllowedFileExtensionsInSampleType test.
+        log("Clear the file field.");
+        FileInput el = FileInput.FileInput(Locator.name("quf_" + attachmentField), getDriver()).findWhenNeeded();
+        executeScript("arguments[0].value = '';", el.getComponentElement());
```

Comment:
Does `WebElement.clear` work for this?
```suggestion
        FileInput el = FileInput.FileInput(Locator.name("quf_" + attachmentField), getDriver()).findWhenNeeded();
        el.getComponentElement().clear();
```

If it does, consider adding a `clear` method to `FileInput`.

Reply:
It appears to. I've added `FileInput.clear`.